modules/storages/spec/features/view_project_storage_members_spec.rb

```diff
@@ -94,4 +76,28 @@
       expect(page).to have_selector("#member-#{principal.id} .status", text: status)
     end
   end
+
+  def create_project_with_storage_and_members
+    role_can_read_files = create(:role, permissions: %i[manage_storages_in_project read_files])
+    role_cannot_read_files = create(:role, permissions: %i[manage_storages_in_project])
+
+    create(:project,
+           members: { user => role_can_read_files,
+                      admin_user => role_cannot_read_files,
+                      connected_user => role_can_read_files,
+                      connected_no_permissions_user => role_cannot_read_files,
+                      disconnected_user => role_can_read_files },
+           enabled_module_names: %i[storages])
+  end
+
+  def create_nextcloud_storage_with_oauth_application
+    oauth_application = create(:oauth_application)
+    create(:nextcloud_storage, :as_automatically_managed, oauth_application:)
+  end
+
+  def create_oauth_client_tokens_for_users(oauth_client:, users:)
+    users.each do |user|
+      create(:oauth_client_token, oauth_client:, user:)
+    end
+  end
```

Comment:
Beautiful. 🖌️

Reply:
🙇🏾  Thanks! Huge fan of inline-helper methods as a replacement for long `let` setups 😄